search/search_service/proxy/es_search_proxy.py

```diff
@@ -0,0 +1,316 @@
+    
+        return filter_queries
+
+
+    def _build_elasticsearch_query(self, *,
+                                   resource: Resource,
+                                   query_term: str,
+                                   filters: List[Filter]) -> Q:
+
+        term_query = self._build_term_query(resource=resource,
+                                            query_term=query_term)
+        filters = self._build_filters(resource=resource, filters=filters)
+
+        es_query = None
+
+        if filters and term_query:
+            es_query = Q('bool', should=[term_query], filter=filters)
+        elif not filters and term_query:
+            es_query = Q('bool', should=[term_query])
+        elif filters and not term_query:
+            es_query = Q('bool', filter=filters)
+        else:
+            raise ValueError("Invalid search query")
+
+        return es_query
+
+
+    def _format_response(self, page_index: int,
+                         results_per_page: int,
+                         responses: List[Response],
+                         resource_types: List[Resource]) -> dict:
+        resource_types_str = [r.name.lower() for r in resource_types]
+        no_results_for_resource = {
+            "results": [],
+            "total_results": 0
+        }
+        results_per_resource = {resource: no_results_for_resource for resource in resource_types_str}
+
+        for r in responses:
+            if r.success():
+                results_count = r.hits.total.value
+                if results_count > 0:
+                    resource_type = r.hits.hits[0]._type
+                    results = []
+                    for search_result in r.hits.hits:
+                        # mapping gives all the fields in the response
+                        result = {}
+                        fields = self.RESOUCE_TO_MAPPING[Resource[resource_type.upper()]]
+                        for f in fields.keys():
+                            # remove "raw" from mapping value
+                            field = fields.get(f).split('.')[0]
+                            result[f] = search_result._source[field]
+                        result["search_score"] = search_result._score
+                        results.append(result)
+                    # replace empty results with actual results
+                    results_per_resource[resource_type] = {
+                        "results": results,
+                        "total_results": results_count
+                    }
+            else:
+                return {
+                    "msg": "Failure",
+                    "page_index": page_index,
+                    "results_per_page": results_per_page,
+                    "results": [],
+                    "status_code": 400,
+                    #  TODO surface actual error
+                    # this makes it so if one resource request fails the whole thing errors
+                    # is this the desired behavior?
```

Comment:
seems correct. And yes logging would be good.

Reply:
I was wondering if it was worth only returning results for the resources that didn't fail but that has the potential for us to have a broken query for a resource and not notice until we're like huh we havent gotten any results for this one particular one idk